modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/serializer/SchedulerSerializer.java

```diff
@@ -125,59 +129,68 @@ private static ScheduledJob editScheduledJob( final ScheduledJobEditor editor, f
 
     private static void addCalendar( final ScheduledJob modifiedJob, final PropertySet data )
     {
-        if ( modifiedJob.getCalendar() != null )
-        {
-            final PropertySet calendarSet = new PropertySet();
+        final PropertySet calendarSet = new PropertySet();
 
-            if ( modifiedJob.getCalendar() instanceof CronCalendar )
-            {
-                final CronCalendar cronCalendar = ( (CronCalendar) modifiedJob.getCalendar() );
+        if ( ScheduleCalendarType.CRON.equals( modifiedJob.getCalendar().getType() ) )
```

Comment:
maybe `if-else if` instead of `if` and `if`?

Reply:
harder to read, don't give any valuable performance improve

Reply:
maybe use switch - as it is natural for enums

Reply:
also, with current implementation new ScheduleCalendarType pass without error.